engine/src/main/java/io/zeebe/engine/processing/deployment/distribute/DeploymentDistributeProcessor.java

```diff
@@ -75,7 +75,13 @@ public void processRecord(
     final DirectBuffer bufferView = new UnsafeBuffer();
     bufferView.wrap(buffer, 0, deploymentEvent.getLength());
 
-    distributeDeployment(key, position, bufferView, streamWriter);
+    // don't distribute the deployment on reprocessing because it may be already distributed
+    // instead, distribute the pending deployments after the reprocessing
```

Comment:
This is not really the reason right. The reason is that we will write concurrently/async a `DISTRIBUTED` on reprocessing which totally breaks the detection.

Reply:
Maybe we add a link to the github issue

Reply:
It is for both reasons :sweat_smile: